Dockerfile

```diff
@@ -0,0 +1,57 @@
+# 1: Build the exe
+FROM rust:1.56 as builder
+
+# WORKDIR /usr/src
+
+# RUN USER=root cargo new agora
+
+# COPY Cargo.toml Cargo.lock /usr/src/agora/
+
+WORKDIR /usr/src/agora
+
+# Prepare for static linking
+# RUN apt-get update && \
+#     apt-get dist-upgrade -y && \
+#     apt-get install -y musl-tools && \
+#     rustup target add x86_64-unknown-linux-musl
+
+# Download and compile Rust dependencies (and store as a separate Docker layer)
+
+# copy local dependencies
+# COPY ./agora-lnd-client ./agora-lnd-client
+# COPY ./lnd-test-context ./lnd-test-context
+# COPY ./bin ./bin
+# build dependencies
+# RUN cargo build --release
```

Comment:
This block of code is a failed attempt to build dependencies first. It resulted in the final binary being a default hello world rust program, not agora.

Reply:
Can't we just build the `agora` binary, and it will build all dependencies automatically?

Reply:
yes, but building everything together means everything has to be rebuilt (including dependencies) every time we change anything in src. If we can build dependencies independently, then we rebuild times are much shorter. I'll remove this for now and make it a TODO

Reply:
Sounds good! Feel free to open issues to track things, I think that's better than including TODOs in the source code.